apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx

```diff
@@ -277,10 +278,18 @@ export default function SectionsSetUpContainer({sectionToBeEdited}) {
         this might mean creating a different button for the "edit" page
         */}
         <Button
-          text={isNewSection ? i18n.finishCreatingSections() : i18n.save()}
+          text={
+            isSaveInProgress
+              ? i18n.saving()
+              : isNewSection
+              ? i18n.finishCreatingSections()
+              : i18n.save()
+          }
           color={Button.ButtonColor.brandSecondaryDefault}
+          disabled={isSaveInProgress}
           onClick={e => {
             e.preventDefault();
+            setIsSaveInProgress(true);
```

Comment:
Just checking: we currently don't have a flow for if the save fails right? So we'd never want `isSaveInProgress` to go back to false?

Reply:
sweet thanks!